From 6ff2c5cd3d4b323d9ce9e306eb2ff23123bc49e7 Mon Sep 17 00:00:00 2001
From: Marcel Kolaja <mkolaja@redhat.com>
Date: Tue, 21 Jan 2014 10:57:08 -0500
Subject: [PATCH 098/112] Honor a symlink when generating configuration by
 grub2-mkconfig

Honor a symlink when generating configuration by grub2-mkconfig, so that
the -o option follows it rather than overwriting it with a regular file.
---
 util/grub-mkconfig.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util/grub-mkconfig.in b/util/grub-mkconfig.in
index ca040dd..b00f9e6 100644
--- a/util/grub-mkconfig.in
+++ b/util/grub-mkconfig.in
@@ -272,7 +272,8 @@ and /etc/grub.d/* files or please file a bug report with
     echo >&2
   else
     # none of the children aborted with error, install the new grub.cfg
-    mv -f ${grub_cfg}.new ${grub_cfg}
+    cat ${grub_cfg}.new > ${grub_cfg}
+    rm -f ${grub_cfg}.new
   fi
 fi
 
-- 
1.8.5.3

